### ISSUE TITLE
Training loss is NaN now.

### ISSUE THREAD

**Issue by Strive21** [PKU-YuanGroup/ConsisID#17]
When will the latest version of the code and data processing code be released?

**Comment by SHYuanBest** [comment]
Thanks for your interest. Is the training loss NaN at the beginning, what dataset did you use? The latest version of the code may not be released so soon. We will prioritize the release of data processing code and the integration of ConsisID into [diffusers](https://github.com/huggingface/diffusers).

**Comment by Strive21** [comment]
> Thanks for your interest. Is the training loss NaN at the beginning, what dataset did you use? The latest version of the code may not be released so soon. We will prioritize the release of data processing code and the integration of ConsisID into [diffusers](https://github.com/huggingface/diffusers).

I downloaded your dataset and processed it appropriately, using CogvideoX-5B-I2V to initialize the weights，which bs is 5 and lr is 3e-7. It has loss in the initial training, but NaN appears after about 500 iterations. Is it because I processed the data wrong? And “fail to detect face using insightface, extract embedding on align face“ occurs during training。

**Comment by SHYuanBest** [comment]
Oh, I see. This may be a problem with `MM-DiT`. The training is very unstable because the activation value of the middle layer may be very large, resulting in loss NaN. You can try turning on `EMA`, `gradient accumulation`, increasing `batchsize`, and reducing the `learning rate`. Another method is to add a `regularization` term to the output of the middle layer.

**Comment by SHYuanBest** [comment]
`fail to detect face using insightface, extract embedding on align face` this warning cannot be avoided because facexlib may not be able to detect the face, and the code will automatically skip this training sample.

**Comment by SHYuanBest** [comment]
or you can try to train only LoRA instead of all parameters.

> Oh, I see. This may be a problem with `MM-DiT`. The training is very unstable because the activation value of the middle layer may be very large, resulting in loss NaN. You can try turning on `EMA`, `gradient accumulation`, increasing `batchsize`, and reducing the `learning rate`. Another method is to add a `regularization` term to the output of the middle layer.

**Comment by SHYuanBest** [comment]
we have release the data processing code

**Comment by SHYuanBest** [comment]
> When will the latest version of the code and data processing code be released?

we have release the data processing code, please refer to [here](https://github.com/PKU-YuanGroup/ConsisID/tree/main/data_preprocess) for more details.

**Comment by tyrink** [comment]
> or you can try to train only LoRA instead of all parameters.
> 
> > Oh, I see. This may be a problem with `MM-DiT`. The training is very unstable because the activation value of the middle layer may be very large, resulting in loss NaN. You can try turning on `EMA`, `gradient accumulation`, increasing `batchsize`, and reducing the `learning rate`. Another method is to add a `regularization` term to the output of the middle layer.
Hi! If train only LoRA instead of all parameters, what is the demanded amount of training data and the number of training iterations?

**Comment by glimmer16** [comment]
> > Thanks for your interest. Is the training loss NaN at the beginning, what dataset did you use? The latest version of the code may not be released so soon. We will prioritize the release of data processing code and the integration of ConsisID into [diffusers](https://github.com/huggingface/diffusers).
> 
> I downloaded your dataset and processed it appropriately, using CogvideoX-5B-I2V to initialize the weights，which bs is 5 and lr is 3e-7. It has loss in the initial training, but NaN appears after about 500 iterations. Is it because I processed the data wrong? And “fail to detect face using insightface, extract embedding on align face“ occurs during training。

Hi! Have you solved this problem? I meet the same issue and wonder which way to avoid loss NaN.

**Comment by SHYuanBest** [comment]
> > > Thanks for your interest. Is the training loss NaN at the beginning, what dataset did you use? The latest version of the code may not be released so soon. We will prioritize the release of data processing code and the integration of ConsisID into [diffusers](https://github.com/huggingface/diffusers).
> > 
> > 
> > I downloaded your dataset and processed it appropriately, using CogvideoX-5B-I2V to initialize the weights，which bs is 5 and lr is 3e-7. It has loss in the initial training, but NaN appears after about 500 iterations. Is it because I processed the data wrong? And “fail to detect face using insightface, extract embedding on align face“ occurs during training。
> 
> Hi! Have you solved this problem? I meet the same issue and wonder which way to avoid loss NaN.

You may need to construct a higher dataset to continue finetuning ConsisID, or have a larger batch size. Since ConsisID is trained on a higher quality internal dataset, if it continues to be trained on the ConsisID-Preview-Data, it is likely to get worse.

**Comment by SHYuanBest** [comment]
Or you can load the ckpt of CogVideoX-5B-I2V for training IPT2V from scratch. (Instead of load ConsisID-Preview for continue finetuning.)

**Comment by SHYuanBest** [comment]
Some solutions can refer to https://github.com/PKU-YuanGroup/ConsisID/issues/31.

**Comment by Strive21** [comment]
> > > Thanks for your interest. Is the training loss NaN at the beginning, what dataset did you use? The latest version of the code may not be released so soon. We will prioritize the release of data processing code and the integration of ConsisID into [diffusers](https://github.com/huggingface/diffusers).
> > 
> > 
> > I downloaded your dataset and processed it appropriately, using CogvideoX-5B-I2V to initialize the weights，which bs is 5 and lr is 3e-7. It has loss in the initial training, but NaN appears after about 500 iterations. Is it because I processed the data wrong? And “fail to detect face using insightface, extract embedding on align face“ occurs during training。
> 
> Hi! Have you solved this problem? I meet the same issue and wonder which way to avoid loss NaN.

I try a larger batch size and solve this problem.

**Comment by Hickey8** [comment]
> `fail to detect face using insightface, extract embedding on align face` this warning cannot be avoided because facexlib may not be able to detect the face, and the code will automatically skip this training sample. `fail to detect face using insightface, extract embedding on align face` 此警告无法避免，因为 FaceXlib 可能无法检测到人脸，并且代码将自动跳过此训练样本。

So the actual training data size engaged in training is smaller than 31947?
